tsfc/kernel_interface/ufc.py

```diff
@@ -248,8 +231,8 @@ def prepare_coordinates(coefficient, name, interior_facet=False):
                                qualifiers=["const"])]
         variable0 = gem.Variable(name+"_0", (size,))
         variable1 = gem.Variable(name+"_1", (size,))
-        expression = (gem.reshape(variable0, shape),
-                      gem.reshape(variable1, shape))
+        expression = (transpose(gem.reshape(variable0, transposed_shape)),
+                      transpose(gem.reshape(variable1, transposed_shape)))
```

Comment:
OK, so previously you treated all coefficients specially except the coordinates.  Now you treat just the coordinates specially.